WickedEngine/wiECS.h

```diff
@@ -423,42 +422,48 @@ namespace wi::ecs
 
 		// Create an instance of ComponentManager of a certain data type
 		// Once added, cannot be removed!
-		template<typename T> inline ComponentManager<T>& Register(uint64_t componentVersion = 0, uint64_t archiveVersion = 0){
+		template<typename T> inline ComponentManager<T>& Register()
+		{
 			componentManagers.push_back(std::make_unique<ComponentManager<T>>());
-			componentVersions.push_back(componentVersion);
-			archiveVersions.push_back(archiveVersion);
 			return static_cast<ComponentManager<T>&>(*componentManagers.back());
 		}
 
 
 		// Bulk operations
 		// Perform deep copy of all the contents of "other" into this
-		inline void Copy(const ComponentLibrary& other){
-			for(size_t i = 0; i < other.componentManagers.size(); ++i){
+		inline void Copy(const ComponentLibrary& other)
+		{
+			for(size_t i = 0; i < other.componentManagers.size(); ++i)
+			{
 				auto& other_compmgr = *other.componentManagers[i];
 				componentManagers[i]->Copy(other_compmgr);
 			}
 		}
 
 		// Check if any component exists for a given entity or not
-		inline bool Contains(Entity entity) const{
+		inline bool Contains(Entity entity) const
+		{
 			bool contains = false;
-			for(auto& componentManager : componentManagers){
+			for(auto& componentManager : componentManagers)
+			{
 				contains = componentManager->Contains(entity);
 				if (contains) break;
 			}
 			return contains;
 		}
 
-		inline void Serialize(wi::Archive& archive, EntitySerializer& seri){
-			for(auto& componentManager : componentManagers){
+		inline void Serialize(wi::Archive& archive, EntitySerializer& seri)
+		{
+			for(auto& componentManager : componentManagers)
+			{
 				if(archive.IsReadMode())
 				{
 					uint64_t archiveVersion;
 					uint64_t componentVersion;
 					archive >> archiveVersion;
 					archive >> componentVersion;
-					if((archive.GetVersion() >= archiveVersion) && (componentVersion >= libraryVersion)){
+					if((archive.GetVersion() >= archiveVersion) && (libraryVersion >= componentVersion))
```

Comment:
I switched the order of checks between componentVersion and libraryVersion. Otherwise base engine that is libraryVersion = 0 would try to load custom components that were created with a libraryVersion > 0.